### DISCUSSION TITLE
Update to PHP 8.3

### DISCUSSION THREAD

**Discussion by denis-korolev** [PHPOffice/PhpSpreadsheet#3985] [Q&A] [answered]
Hi, a want to update my project to PHP 8.3 but phpoffice/phpspreadsheet php:   is php ^8.0

Can you update it to 8.3 ?

**Comment by MarkBaker** [comment, marked as answer]
`^8.0` includes PHP 8.3. `^8.0` means >=8.0 and <9.0.
See [Composer Documentation](https://getcomposer.org/doc/articles/versions.md#caret-version-range-) for an explanation of the Composer `^` symbol in version constraints

**Comment by cengizhan** [comment]
`E:\code\php\composer>composer require phpoffice/phpspreadsheet
./composer.json has been updated
Running composer update phpoffice/phpspreadsheet
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - laminas/laminas-escaper 2.10.0 requires php ^7.4 || ~8.0.0 || ~8.1.0 -> your php version (8.3.6) does not satisfy that requirement.
    - phpoffice/phpword 0.18.3 requires laminas/laminas-escaper ^2.2 -> satisfiable by laminas/laminas-escaper[2.10.0].
    - phpoffice/phpword is locked to version 0.18.3 and an update of this package was not requested.

You can also try re-running composer require with an explicit version constraint, e.g. "composer require phpoffice/phpspreadsheet:*" to figure out if any version is installable, or "composer require phpoffice/phpspreadsheet:^2.1" if you know which you need.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.`

**Reply by oleibman** [reply]
You need to get yourself a current version of phpword. Among other considerations, the current release no longer has a dependency on laminas-escaper.

**Reply by cengizhan** [reply]
thanks. there were some leftovers in composer.json. deleting them fixed
